drivers/spi/spi_stm32lx.c

```diff
@@ -0,0 +1,398 @@
+	struct spi_stm32lx_data *data = DEV_DATA(dev);
+	volatile u8_t *dr8 = &spi->dr.dr_as_u8;
+
+	if (spi->sr.bit.txe && data->current.tx_len) {
+		*dr8 = *data->current.tx_buf;
+		data->current.tx_buf++;
+		data->current.tx_len--;
+	}
+
+	if (spi->sr.bit.rxne && data->current.rx_len) {
+		*data->current.rx_buf = *dr8;
+		data->current.rx_buf++;
+		data->current.rx_len--;
+	}
+
+	if (!data->current.rx_len && !data->current.tx_len) {
+		spi->cr2.bit.rxneie = 0;
+		spi->cr2.bit.txeie = 0;
+
+		k_sem_give(&data->device_sync_sem);
+	}
+}
+#endif
+
+#define BR_SCALER_COUNT		8
+static const u32_t baud_rate_scaler[] = {
+	2, 4, 8, 16, 32, 64, 128, 256
+};
+
+/* Only support 8bit data size */
+#define DATA_SIZE_8_CFG		(3 + 4)
+
+static int spi_stm32lx_configure(struct device *dev, struct spi_config *config)
+{
+	volatile struct spi_stm32lx *spi = SPI_STRUCT(dev);
+	struct spi_stm32lx_data *data = DEV_DATA(dev);
+	const struct spi_stm32lx_config *cfg = DEV_CFG(dev);
+	u32_t flags = config->config;
+	u32_t clock;
+	int br;
+
+	clock_control_get_rate(data->clock,
+			(clock_control_subsys_t *)&cfg->pclken, &clock);
+
+	if (SPI_WORD_SIZE_GET(flags) != 8) {
+		return -ENOTSUP;
+	}
+
+	for (br = 0 ; br < BR_SCALER_COUNT ; ++br) {
+		u32_t clk = clock / baud_rate_scaler[br];
+
+		if (clk < config->max_sys_freq) {
+			break;
+		}
+	}
+
+	if (br >= BR_SCALER_COUNT) {
+		return -ENOTSUP;
+	}
+
+	/* Disable Peripheral */
+	spi->cr1.bit.spe = 0;
+
+	/* Setup baud rate prescaler */
+	spi->cr1.bit.br = br;
+
+	/* Setup polarity flags */
+	if (flags & SPI_MODE_CPOL) {
+		spi->cr1.bit.cpol = 1;
+	} else {
+		spi->cr1.bit.cpol = 0;
+	}
+
+	if (flags & SPI_MODE_CPHA) {
+		spi->cr1.bit.cpha = 1;
+	} else {
+		spi->cr1.bit.cpha = 0;
+	}
+
+	/* Full Duplex 2-lines */
+	spi->cr1.bit.rxonly = 0;
+	spi->cr1.bit.bidimode = 0;
+
+	/* Setup transfer bit mode */
+	if (flags & SPI_TRANSFER_MASK) {
+		spi->cr1.bit.lsbfirst = 1;
+	} else {
+		spi->cr1.bit.lsbfirst = 0;
+	}
+
+	/* Disable CRC Feature */
+	spi->cr1.bit.crcen = 0;
+
+	/* Slave Support */
+	if (flags & STM32LX_SPI_SLAVE_MODE) {
+		spi->cr1.bit.mstr = 0;
+
+		/* NSS Management */
+		if (flags & STM32LX_SPI_SLAVE_NSS_IGNORE) {
+			spi->cr1.bit.ssm = 1;
+			spi->cr1.bit.ssi = 0;
+		} else {
+			spi->cr1.bit.ssm = 0;
+		}
+
+		data->current.is_slave = 1;
+	} else {
+		spi->cr1.bit.mstr = 1;
+		spi->cr1.bit.ssm = 0;
+
+		/* NSS Management */
+		if (flags & STM32LX_SPI_MASTER_NSS_IGNORE) {
+			spi->cr2.bit.ssoe = 0;
+		} else {
+			spi->cr2.bit.ssoe = 1;
+			spi->cr2.bit.nssp = 1;
+		}
+
+		data->current.is_slave = 0;
+	}
+
+	/* Setup Data size */
+	spi->cr2.bit.ds = DATA_SIZE_8_CFG;
+	spi->cr2.bit.frxth = 1;
+
+	/* Motorola Format */
+	spi->cr2.bit.frf = 0;
+
+	data->current.rx_len = 0;
+	data->current.rx_buf = NULL;
+	data->current.tx_len = 0;
+	data->current.tx_buf = NULL;
+
+	return 0;
+}
+
+static int spi_stm32lx_slave_select(struct device *dev, u32_t slave)
+{
+	/* NOP */
+
+	return 0;
+}
```

Comment:
you can remove that function altogether and not fill in a value in api_funcs, if NULL the public api will return 0